cpp/src/centrality/katz_centrality.cu

```diff
@@ -39,6 +39,8 @@ gdf_error gdf_katz_centrality(gdf_graph *graph,
     return err;
   GDF_REQUIRE(graph->adjList->offsets->dtype == GDF_INT32, GDF_UNSUPPORTED_DTYPE);
   GDF_REQUIRE(graph->adjList->indices->dtype == GDF_INT32, GDF_UNSUPPORTED_DTYPE);
+  GDF_REQUIRE(katz_centrality->dtype == GDF_FLOAT64, GDF_UNSUPPORTED_DTYPE);
```

Comment:
This is not really a limitation of the implementation. Rather it was a design decision to ensure that the results are more accurate due to insatiability of floating point. Specifically, if float32 is used instead of float64\double, then the conversion might happen in too few iterations to ensure correctness. However, it could be that it might be enough.